core/src/main/java/org/apache/calcite/rel/rules/PruneEmptyRules.java

```diff
@@ -551,15 +554,13 @@ public interface JoinRightEmptyRuleConfig extends PruneEmptyRule.Config {
     }
   }
 
-  private static Stream<RexNode> getProjectStream(
-      RelDataType inRowType,
-      RelDataType castRowType,
-      int castRowTypeOffset,
-      RexBuilder rexBuilder) {
-    return inRowType.getFieldList().stream().map(
-        typeField -> rexBuilder.makeCast(
-                castRowType.getFieldList().get(castRowTypeOffset + typeField.getIndex()).getType(),
-                rexBuilder.makeInputRef(typeField.getType(), typeField.getIndex())));
+  private static Stream<RexNode> castToNullable(
+      RexBuilder rexBuilder, Stream<RexNode> leftExpressions) {
```

Comment:
We probably want to use `expressions` instead of `leftExpressions` here, since the method can be called on left or right, depending on the kind of outer join.

Reply:
fixed